lib/assert.js

```diff
@@ -143,6 +143,12 @@ assert.deepStrictEqual = function deepStrictEqual(actual, expected, message) {
 };
 
 function _deepEqual(actual, expected, strict) {
+  // If both values are instances of typed arrays, wrap them in
+  // a Buffer each to increase performance
+  if (ArrayBuffer.isView(actual) && ArrayBuffer.isView(expected)) {
```

Comment:
Can this logic be pushed down further? At least past the "7.1." check on line 152.